### PR TITLE
feat(route): Update route OpenAI

### PR DESCRIPTION
**Pull request by StevenRCE0**
## 完整路由地址 / Example for the proposed route(s)

```routes
/openai/blog
/openai/blog/events
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

- 修复了该失效路由
- 增加 radar 匹配规则
- 现在通过 API 来获取文章元数据（包括题目分类时间和头图）
- 从全文页面解析作者列表
- 通过模板渲染头图和正文

### CODE REVIEW

File: lib/v2/openai/blog.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    description: article,
+                    pubDate: attributes.createdAt,
+                    category: attributes.tags.map((tag) => tag.title),
+                    link: textUrl,
+                };
+            });
+        })
+    );
+
+    const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1);
```
Comment (TonyRL) [inline lib/v2/openai/blog.js:80]: You can use `toTitleCase`.
https://github.com/DIYgod/RSSHub/blob/48570ccda7ac5c7ce73c5944e522a1f456bbb90b/lib/utils/common-utils.js#L4-L9

```js
const { toTitleCase } = require('@/utils/common-utils');
console.log(toTitleCase(tag))
```

File: lib/v2/openai/blog.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    let apiBaseUrl = initResponse.data
+        .toString()
+        .match(/(?<=TWILL_API_BASE:").+?(?=")/)[0]
+        .replaceAll('\\u002F', '/');
+    apiBaseUrl = apiBaseUrl + '/api/v1/blog-details';
+    const apiUrl = new URL(apiBaseUrl);
+
+    // Construct API query
+    apiUrl.searchParams.append('sort', '-publicationDate,-createdAt');
+    apiUrl.searchParams.append('page[size]', '5');
```
Comment (TonyRL) [inline lib/v2/openai/blog.js:28]:
```suggestion
    apiUrl.searchParams.append('page[size]', '20');
```

Comment (StevenRCE0) [reply lib/v2/openai/blog.js:28]: > ```suggestion
> 
>     apiUrl.searchParams.append('page[size]', '20');
> 
> ```
> 
> They use 20 by default.

True, I thought it's recommended to reduce the number of entries we get for performance purposes (the list was sliced to 5 previously).

File: lib/v2/openai/radar.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+    'openai.com': {
+        _name: 'OpenAI',
+        '.': [
+            {
+                title: 'OpenAI Blog',
```
Comment (TonyRL) [inline lib/v2/openai/radar.js:6]:
```suggestion
                title: 'Blog',
```

**Pull request merged**